src/ripple/overlay/impl/PeerImp.h

```diff
@@ -127,8 +128,8 @@ class PeerImp
     LedgerIndex maxLedger_ = 0;
     uint256 closedLedgerHash_;
     uint256 previousLedgerHash_;
-    std::list<uint256> recentLedgers_;
-    std::list<uint256> recentTxSets_;
+    std::deque<uint256> recentLedgers_;
```

Comment:
Sorry, I was not at all clear.  :-)  

By "policy" I meant, "Some place written down where people can go when they want to choose a C++ container in future."

Right now, if we ignore maps, we have the following general purpose containers which appear in our code:

* `std::vector`
* `std::deque`
* `std::set`
* `ripple::hash_set`
* `ripple::hardened_hash_set`
* `std::unordered_set`
* `std::list`
* `boost::ptr_vector`

and the last three are deprecated - I think...

I know a great deal about these, but there are details I'm not sure of.

For example, I know that `std::set` is "cheaper" (faster and smaller) than `ripple::hash_set` if we aren't expecting "very many" elements - but I don't even have a close order of magnitude idea of how many that might be (8 items?  64 items? 512 items?)

And, when should we be using hardened collections?  On all "dirty" data from the outside world?  On a specific subclass of problems?

Even just a page somewhere summarizing this - perhaps in `basics/README.md` and a reference to it in `basics/UnorderedContainers.h` - might clarify all our minds as to the details.

Or I could do this as a followup!